### ISSUE TITLE
Function fails when it is run in parallel

### ISSUE THREAD

**Issue by lyriccoder** [noxdafox/pebble#74]
I have the following code:
```
# files_without_tests = list of Java files

results = []
    with ProcessPool(system_cores_qty) as executor:
        tasks = [TaskAggregatorJavaFiles(file=str(x)) for x in files_without_tests]
        future = executor.map(d6tflow.run, tasks, timeout=200, )
        result = future.result()

        for filename in tqdm(files_without_tests):
            try:
                next(result)
                data = TaskAggregatorJavaFiles(file=str(filename)).outputLoad()
                if data:
                    results.append(data)
            except Exception as e:
                print(e)
        print(results)
```
The `d6tflow.run` function can't be run in parallel 

```===== Luigi Execution Summary =====
Traceback (most recent call last):
  File "/home/hm/.local/lib/python3.8/site-packages/pebble/pool/process.py", line 408, in worker_process
    send_result(channel, Result(task.id, result))
  File "/home/hm/.local/lib/python3.8/site-packages/pebble/pool/process.py", line 408, in worker_process
    send_result(channel, Result(task.id, result))
Traceback (most recent call last):
Traceback (most recent call last):
  File "/home/hm/.local/lib/python3.8/site-packages/pebble/pool/process.py", line 408, in worker_process
    send_result(channel, Result(task.id, result))

  File "/home/hm/.local/lib/python3.8/site-packages/pebble/pool/process.py", line 408, in worker_process
    send_result(channel, Result(task.id, result))
  File "/home/hm/.local/lib/python3.8/site-packages/pebble/pool/process.py", line 425, in send_result
    pipe.send(result)
  File "/home/hm/.local/lib/python3.8/site-packages/pebble/pool/process.py", line 425, in send_result
    pipe.send(result)
  File "/usr/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/home/hm/.local/lib/python3.8/site-packages/pebble/pool/process.py", line 425, in send_result
    pipe.send(result)
  File "/home/hm/.local/lib/python3.8/site-packages/pebble/pool/process.py", line 425, in send_result
    pipe.send(result)
  File "/home/hm/.local/lib/python3.8/site-packages/pebble/pool/channel.py", line 100, in unix_send
    return self.writer.send(obj)
Process Process-4:
  File "/home/hm/.local/lib/python3.8/site-packages/pebble/pool/channel.py", line 100, in unix_send
    return self.writer.send(obj)
  File "/usr/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/home/hm/.local/lib/python3.8/site-packages/pebble/pool/channel.py", line 100, in unix_send
    return self.writer.send(obj)
  File "/home/hm/.local/lib/python3.8/site-packages/pebble/pool/channel.py", line 100, in unix_send
    return self.writer.send(obj)
  File "/usr/lib/python3.8/multiprocessing/connection.py", line 206, in send
    self._send_bytes(_ForkingPickler.dumps(obj))
  File "/usr/lib/python3.8/multiprocessing/connection.py", line 206, in send
    self._send_bytes(_ForkingPickler.dumps(obj))
  File "/home/hm/.local/lib/python3.8/site-packages/pebble/pool/process.py", line 410, in worker_process
    os._exit(error.errno if error.errno else 1)
  File "/home/hm/.local/lib/python3.8/site-packages/pebble/pool/process.py", line 410, in worker_process
    os._exit(error.errno if error.errno else 1)
  File "/usr/lib/python3.8/multiprocessing/connection.py", line 206, in send
    self._send_bytes(_ForkingPickler.dumps(obj))
  File "/usr/lib/python3.8/multiprocessing/connection.py", line 206, in send
    self._send_bytes(_ForkingPickler.dumps(obj))
  File "/usr/lib/python3.8/multiprocessing/reduction.py", line 51, in dumps
    cls(buf, protocol).dump(obj)
  File "/usr/lib/python3.8/multiprocessing/reduction.py", line 51, in dumps
    cls(buf, protocol).dump(obj)
AttributeError: 'RuntimeError' object has no attribute 'errno'
AttributeError: 'RuntimeError' object has no attribute 'errno'```

Could you please tell me why it fails? Usually, `ProcessPool` works good with other functions, but if fails if I pass `d6tflow.run`

**Comment by noxdafox** [comment]
Hard to say without looking at the original function.

The error suggests a pickling error when trying to return something to the main loop. This probably indicates the value `d6tflow.run` is trying to return is not [picklable](https://docs.python.org/3/library/pickle.html#what-can-be-pickled-and-unpickled).

**Comment by lyriccoder** [comment]
Does the pebble has another serializer?

**Comment by noxdafox** [comment]
Pebble relies on Python standard library and therefore uses Pickle internally.

If you want to use third-party serializers (such as [dill](https://pypi.org/project/dill/) for example) you need to handle the serialization and de-serialization yourself.

**Comment by noxdafox** [comment]
Improved documentation clarifications and fixed error reporting in `4.6.1`.
